events/apigw.go

```diff
@@ -133,63 +133,65 @@ type APIGatewayV2HTTPResponse struct {
-	Authorizer         interface{}               `json:"authorizer"`
-	HTTPMethod         string                    `json:"httpMethod"`
-	APIID              string                    `json:"apiId"` // The API Gateway rest API Id
-	ConnectedAt        int64                     `json:"connectedAt"`
-	ConnectionID       string                    `json:"connectionId"`
-	DomainName         string                    `json:"domainName"`
-	Error              string                    `json:"error"`
-	EventType          string                    `json:"eventType"`
-	ExtendedRequestID  string                    `json:"extendedRequestId"`
-	IntegrationLatency string                    `json:"integrationLatency"`
-	MessageDirection   string                    `json:"messageDirection"`
-	MessageID          interface{}               `json:"messageId"`
-	RequestTime        string                    `json:"requestTime"`
-	RequestTimeEpoch   int64                     `json:"requestTimeEpoch"`
-	RouteKey           string                    `json:"routeKey"`
-	Status             string                    `json:"status"`
+	AccountID            string                    `json:"accountId,omitempty"`
+	ResourceID           string                    `json:"resourceId,omitempty"`
+	Stage                string                    `json:"stage"`
+	RequestID            string                    `json:"requestId"`
+	Identity             APIGatewayRequestIdentity `json:"identity"`
+	ResourcePath         string                    `json:"resourcePath,omitempty"`
+	Authorizer           interface{}               `json:"authorizer,omitempty"`
+	HTTPMethod           string                    `json:"httpMethod,omitempty"`
+	APIID                string                    `json:"apiId"` // The API Gateway rest API Id
+	ConnectedAt          int64                     `json:"connectedAt"`
+	ConnectionID         string                    `json:"connectionId"`
+	DomainName           string                    `json:"domainName"`
+	Error                string                    `json:"error,omitempty"`
+	EventType            string                    `json:"eventType"`
+	ExtendedRequestID    string                    `json:"extendedRequestId"`
+	IntegrationLatency   string                    `json:"integrationLatency,omitempty"`
+	MessageDirection     string                    `json:"messageDirection"`
+	MessageID            *string                   `json:"messageId,omitempty"`
+	RequestTime          string                    `json:"requestTime"`
+	RequestTimeEpoch     int64                     `json:"requestTimeEpoch"`
+	RouteKey             string                    `json:"routeKey"`
+	Status               string                    `json:"status,omitempty"`
+	DisconnectStatusCode int64                     `json:"disconnectStatusCode,omitempty"`
+	DisconnectReason     *string                   `json:"disconnectReason,omitempty"`
```

Comment:
I couldn't find a reference to these fields in the AWS Docs, nor in the event libraries for .NET or Java. 

If you happen to have a reference handy? My search skills are failing me right now.

Reply:
Hi, I found this sample online, you can also recreate this by looking into the websocket api gateway log on cloudwatch. I couldn't find anything in the aws documents neither. 
```
    Host: '3xcls223mg.execute-api.eu-west-1.amazonaws.com',
    'x-api-key': '',
    'X-Forwarded-For': '',
    'x-restapi': ''
  },
  multiValueHeaders: {
    Host: [ '3xcls223mg.execute-api.eu-west-1.amazonaws.com' ],
    'x-api-key': [ '' ],
    'X-Forwarded-For': [ '' ],
    'x-restapi': [ '' ]
  },
  requestContext: {
    routeKey: '$disconnect',
    disconnectStatusCode: 1006,
    eventType: 'DISCONNECT',
    extendedRequestId: 'fvLJ9EMQDoEFXSw=',
    requestTime: '22/May/2021:15:38:26 +0000',
    messageDirection: 'IN',
    disconnectReason: 'Connection Closed Abnormally',
    stage: 'updatedevice',
    connectedAt: 1621697605538,
    requestTimeEpoch: 1621697906882,
    identity: {
      userAgent: 'arduino-WebSocket-Client',
      sourceIp: '137.97.106.23'
    },
    requestId: 'fvLJ9EMQDoEFXSw=',
    domainName: '3xcls223mg.execute-api.eu-west-1.amazonaws.com',
    connectionId: 'fvKa4d88joECFDw=',
    apiId: '3xcls223mg'
  },
  isBase64Encoded: false
}
```
Link to the sample: https://github.com/Links2004/arduinoWebSockets/issues/667

Reply:
Also found here: https://github.com/boto/boto3/issues/3789

Reply:
Unfortunately I'll need a reference that comes from docs.aws.amazon.com

I'll otherwise make time to walk myself through the [developer guide](https://docs.aws.amazon.com/apigateway/latest/developerguide/apigateway-websocket-api.html) so that I can reproduce the test payloads in this PR.

Reply:
I found this developer guide updated!

https://docs.aws.amazon.com/apigateway/latest/developerguide/apigateway-websocket-api-integration-requests.html#api-gateway-simple-proxy-for-lambda-input-format-websocket

Reply:
Is there anything else I can do?

Reply:
That's perfect! Thanks for following up!